test/api_integration/apis/ui_counters/ui_counters.ts

```diff
@@ -32,7 +32,7 @@ export default function ({ getService }: FtrProviderContext) {
       .expect(200);
 
     // wait for SO to index data into ES
-    await new Promise((res) => setTimeout(res, 5 * 1000));
+    await new Promise((res) => setTimeout(res, 8 * 1000));
```

Comment:
Can we retry requesting `/api/saved_objects/_find?type=usage-counters` until it's succeeded instead of relying on a manual delay?

Reply:
That is doable although the delay pattern is all across kibana tests (200+ places)

Reply:
Agree, but it doesn't mean it's a good thing.
- it leads to false positives due to changes in Kibana internals (what you are fixing right now)
- it introduces unnecessary delays in the execution flow. Maybe the condition is true after 2sec? 
- it's not clear how to pick a correct value, devs tend to copy-paste them

Reply:
I completely agree. What i meant is that we might need to resolve it across all tests rather than just this one test. I've created a helper util (with tests) to retry assertions. We can move it outside the `tests/.../telemetry` folder in a separate PR later on. Let me know what you think.

Reply:
>  We can move it outside the tests/.../telemetry folder in a separate PR later on.

ok then. Let's create an issue for Core-related tests.